src/Azure/Akka.Streams.Azure.EventHub.V5/EventHubSource.cs

```diff
@@ -78,28 +87,30 @@ private void OnProcessEvents((TaskCompletionSource<Task<Done>>, ProcessContext)
                         context.Partition.PartitionId,
                         context.Partition.ConsumerGroup,
                         context.Partition.EventHubName,
-                        context.Data?.Offset);
+                        context.Data?.Offset.ToString() ?? "null");
                 }
 
-                _pendingEvent = context;
-                _currentProcessingTask = new TaskCompletionSource<Done>();
+                var holder = new Holder(context);
+                _pendingQueue.Enqueue(holder);
```

Comment:
Fix, on multi partition EventHub, `ProcessEventAsync` is called concurrently for every PartitionId in the ProcessorClient. The old code assumes that the processor will wait for every async call, now we buffer all async calls and completes each task when the event is successfully pushed down the stream.